### PR TITLE
Draft nfc editor version 1

### PR DESCRIPTION
**Pull request by LionZXY**
**Background**

We want edit NFC card content

**Changes**

- Add NFC editor sample
- Add autoresize for editor
- Add key input bus

**Test plan**

Try open sample

### CODE REVIEW

File: components/core/ui/fragment/src/main/java/com/flipperdevices/core/ui/fragment/ComposeFragment.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ abstract class ComposeFragment : Fragment(), StatusBarColorProvider {
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return ComposeView(requireContext()).apply {
+        return FlipperComposeView(requireContext()).apply {
```
Comment (Programistich) [inline components/core/ui/fragment/src/main/java/com/flipperdevices/core/ui/fragment/ComposeFragment.kt:29]: We use FlipperComposeView on all app. Do we need all the functionality of the new fragment for the application?

File: components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * 3) Use an invisible character from https://invisible-characters.com/
+ * 4) Move cursor from 0 position to previous input: [00][|00] -> [00|][00]
+ *
+ * But all methods have disadvantages. Either for the user or for the architecture.
+ * We chose the dirtiest way for the architecture, but which works best and most stable.
+ *
+ * There may still be good ways to do this in the future:
+ * https://twitter.com/ychescale9/status/1372216269542334464
+ */
+interface KeyInputBus {
```
Comment (Programistich) [inline components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt:23]: api and impl in one module. Any reason?

File: components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+@Composable
+fun ComposableNfcEditor(nfcEditorViewModel: NfcEditorViewModel = viewModel()) {
+    BoxWithConstraints(Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+        CompositionLocalProvider(
+            LocalTextStyle provides TextStyle(
+                fontWeight = FontWeight.W500,
+                fontSize = 14.sp,
+                color = LocalPallet.current.text100,
+                fontFamily = FontFamily(
+                    Font(
```
Comment (Programistich) [inline components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt:32]: Why not setup font on FlipperTypography?

File: components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcLine.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    onFocusChanged: (NfcEditorCellLocation, isFocused: Boolean) -> Unit = { _, _ -> },
+    onValueChanged: (NfcEditorCellLocation, TextFieldValue) -> Unit = { _, _ -> }
+) {
+    Row {
+        Text(
+            modifier = Modifier.width((scaleFactor * WIDTH_LINE_INDEX_DP).dp),
+            text = lineIndex.toString(),
+            textAlign = TextAlign.End,
+            style = LocalTextStyle.current.copy(
+                color = LocalPallet.current.text16,
```
Comment (Programistich) [inline components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcLine.kt:37]: NIT: We can don't use copy. In bottom write style = LocalTextStyle.current, color = LocalPallet.current.text16, fontSize = (scaleFactor * FONT_SIZE_SP).sp

**Review by Programistich — CHANGES_REQUESTED** [review]

File: components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt (hunk 2 above)
Comment (LionZXY) [reply components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt:23]: Modules in core are always used in this logic. Do you think we should break it down into api and impl?

File: components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt (hunk 3 above)
Comment (LionZXY) [reply components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt:32]: Can you provide an example? I don't understand what it's supposed to look like. I would appreciate it.

File: components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcLine.kt (hunk 4 above)
Comment (LionZXY) [reply components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcLine.kt:37]: Exactly!

File: components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt (hunk 3 above)
Comment (Programistich) [reply components/nfceditor/impl/src/main/java/com/flipperdevices/nfceditor/impl/composable/ComposableNfcEditor.kt:32]: in data class FlipperTypograhy create new field with this font
and use like LocalTypography.current.{your field}

File: components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt (hunk 2 above)
Comment (Programistich) [reply components/core/keyinputbus/src/main/java/com/flipperdevices/core/keyinputbus/KeyInputBus.kt:23]: I think its NIT

**Pull request merged**